app/utils/file/index.ts

```diff
@@ -365,3 +371,71 @@ export function getLocalFilePathFromFile(dir: string, serverUrl: string, file: F
 
     return undefined;
 }
+
+export async function extractFileInfos(files: Array<Asset | DocumentPickerResponse | PastedFile>) {
+    const out: ExtractedFileInfo[] = [];
+
+    await Promise.all(files.map(async (file) => {
+        if (!file) {
+            return;
+        }
+
+        const outFile = {
+            progress: 0,
+            localPath: file.uri,
+            clientId: generateId(),
+            loading: true,
+        } as unknown as ExtractedFileInfo;
+
+        if (file.hasOwnProperty('fileSize')) {
```

Comment:
If you use the `in` operator, you may not need to do the casting:
https://www.typescriptlang.org/docs/handbook/2/narrowing.html#the-in-operator-narrowing

Reply:
Thanks, I have added it.